brushfire-core/src/main/scala/com/stripe/brushfire/Predicate.scala

```diff
@@ -1,25 +1,59 @@
 package com.stripe.brushfire
 
-sealed trait Predicate[V] {
-  def apply(v: Option[V]): Boolean
-}
+/**
+ * A `Predicate` is a function given a possibly missing feature value returns
+ * `true` or `false`. It is generally used within a [[Tree]] to decide which
+ * branch to follow while trying to classify a row/feature vector.
+ */
+sealed trait Predicate[V] extends (Option[V] => Boolean)
 
+/**
+ * A [[Predicate]] that returns `true` iff the input is missing or the input is
+ * defined and is equal to `value` (according to the input's `equals` method).
+ */
 case class EqualTo[V](value: V) extends Predicate[V] {
-  def apply(v: Option[V]) = v.isDefined && (v.get == value)
+  def apply(v: Option[V]) = v.isEmpty || (v.get == value)
 }
 
+/**
+ * A [[Predicate]] that returns `true` iff the input is missing or if the input
+ * is defined and it is *less than* `value`. This uses the implicit `Ordering`
+ * of type `V` to handle the comparison.
+ */
 case class LessThan[V](value: V)(implicit ord: Ordering[V]) extends Predicate[V] {
-  def apply(v: Option[V]) = !v.isDefined || ord.lt(v.get, value)
+  def apply(v: Option[V]) = v.isEmpty || ord.lt(v.get, value)
 }
 
+/**
+ * A [[Predicate]] that returns `true` if `pred` returns `false` and returns
+ * `false` if `pred` returns `true`.
+ */
 case class Not[V](pred: Predicate[V]) extends Predicate[V] {
-  def apply(v: Option[V]) = !pred(v)
+  def apply(v: Option[V]) = v.isEmpty || !pred(v)
 }
 
+/**
+ * A [[Predicate]] that returns `true` if any of the predicates in `preds`
+ * returns `true`.
+ */
 case class AnyOf[V](preds: Seq[Predicate[V]]) extends Predicate[V] {
   def apply(v: Option[V]) = preds.exists { p => p(v) }
 }
 
+/**
+ * A [[Predicate]] that will only return `true` if, at least, the value is
+ * defined (not missing). Normally, predicates will treat a missing value (an
+ * input of `None`) as a success and return `true`, but this is not always
+ * desired. `IsPresent` allows an additional check that *requires* the value
+ * be present to succeed.
+ *
+ * If `pred` is `None`, then this is a predicate that returns `true` iff the
+ * value is present (not missing).
+ */
+case class IsPresent[V](pred: Option[Predicate[V]]) extends Predicate[V] {
```

Comment:
I was wondering whether this is better than a simple `IsPresent` that doesn't wrap anything but that gets and-ed with another predicate ... which I guess we have to construct with de morgan's law ... which, I *think* screws us up because of the default missing feature handling? So I guess this is fine. It's certainly easier to use :)